web-registry/src/components/templates/ProjectDetails/ProjectDetails.loader.ts

```diff
@@ -0,0 +1,83 @@
+import { QueryClient } from '@tanstack/react-query';
+
+import { graphqlClient } from 'lib/clients/graphqlClient';
+import { getEcocreditQueryClient } from 'lib/clients/regen/ecocredit/ecocreditQueryClient';
+import { getMarketplaceQueryClient } from 'lib/clients/regen/ecocredit/marketplace/marketplaceQueryClient';
+import { getProjectQuery } from 'lib/queries/react-query/ecocredit/getProjectQuery/getProjectQuery';
+import { getAllowedDenomQuery } from 'lib/queries/react-query/ecocredit/marketplace/getAllowedDenomQuery/getAllowedDenomQuery';
+import { getSellOrdersExtendedQuery } from 'lib/queries/react-query/ecocredit/marketplace/getSellOrdersExtendedQuery/getSellOrdersExtendedQuery';
+import { getProjectByOnChainIdQuery } from 'lib/queries/react-query/registry-server/graphql/getProjectByOnChainIdQuery/getProjectByOnChainIdQuery';
+import { getAllCreditClassesQuery } from 'lib/queries/react-query/sanity/getAllCreditClassesQuery/getAllCreditClassesQuery';
+import { getAllProjectPageQuery } from 'lib/queries/react-query/sanity/getAllProjectPageQuery/getAllProjectPageQuery';
+import { getFromCacheOrFetch } from 'lib/queries/react-query/utils/getFromCacheOrFetch';
+
+import { client as sanityClient } from '../../../sanity';
+import { getIsOnChainId } from './ProjectDetails.utils';
+
+type LoaderType = {
+  queryClient: QueryClient;
+};
+
+export const projectDetailsLoader =
+  ({ queryClient }: LoaderType) =>
+  async ({ params: { projectId } }: { params: { projectId?: string } }) => {
+    // Params
+    const isOnChainId = getIsOnChainId(projectId);
+
+    // Clients
+    const marketplaceClient = await getMarketplaceQueryClient();
+    const ecocreditClient = await getEcocreditQueryClient();
+
+    // Queries
+    const allProjectPageQuery = getAllProjectPageQuery({ sanityClient });
+    const allCreditClassesQuery = getAllCreditClassesQuery({ sanityClient });
+    const projectQuery = getProjectQuery({
+      request: { projectId },
+      client: ecocreditClient,
+    });
+    const projectByOnChainIdQuery = getProjectByOnChainIdQuery({
```

Comment:
why not also use `getProjectByHandleQuery` in here?

Reply:
Good point, I added it.